examples/regression/1-quickstart/plot_heteroscedastic_1d_data.py

```diff
@@ -0,0 +1,197 @@
+
+    Returns
+    -------
+    Tuple[NDArray, NDArray, NDArray, NDArray, float]
+        Generated training and test data.
+        [0]: X_train
+        [1]: y_train
+        [2]: X_true
+        [3]: y_true
+        [4]: y_true_sigma
+    """
+    np.random.seed(59)
+    q95 = scipy.stats.norm.ppf(0.95)
+    X_train = np.linspace(0, 1, n_train)
+    X_true = np.linspace(0, 1, n_true)
+    y_train = np.array(
+        [
+            (f(x) + (np.random.normal(0, sigma))*x) for x in X_train
+        ]
+    )
+    y_true = f(X_true)
+    y_true_sigma = q95 * sigma
+    return X_train, y_train, X_true, y_true, y_true_sigma
+
+
+def plot_1d_data(
+    X_train: NDArray,
+    y_train: NDArray,
+    X_test: NDArray,
+    y_test: NDArray,
+    y_test_sigma: float,
+    y_pred: NDArray,
+    y_pred_low: NDArray,
+    y_pred_up: NDArray,
+    ax: plt.Axes,
+    title: str,
+) -> None:
+    """
+    Generate a figure showing the training data and estimated
+    prediction intervals on test data.
+
+    Parameters
+    ----------
+    X_train : NDArray
+        Training data.
+    y_train : NDArray
+        Training labels.
+    X_test : NDArray
+        Test data.
+    y_test : NDArray
+        True function values on test data.
+    y_test_sigma : float
+        True standard deviation.
+    y_pred : NDArray
+        Predictions on test data.
+    y_pred_low : NDArray
+        Predicted lower bounds on test data.
+    y_pred_up : NDArray
+        Predicted upper bounds on test data.
+    ax : plt.Axes
+        Axis to plot.
+    title : str
+        Title of the figure.
+    """
+    ax.set_xlabel("x")
+    ax.set_ylabel("y")
+    ax.set_xlim([0, 1])
+    ax.set_ylim([0, 1])
+    ax.scatter(X_train, y_train, color="red", alpha=0.3, label="training")
+    ax.plot(X_test, y_test, color="gray", label="True confidence intervals")
+    ax.plot(X_test, y_test - y_test_sigma, color="gray", ls="--")
+    ax.plot(X_test, y_test + y_test_sigma, color="gray", ls="--")
+    ax.plot(X_test, y_pred, label="Prediction intervals")
+    ax.fill_between(X_test, y_pred_low, y_pred_up, alpha=0.3)
+    ax.set_title(title)
+    ax.legend()
+
+
+X_train, y_train, X_test, y_test, y_test_sigma = get_heteroscedastic_data()
+
+polyn_model = Pipeline(
+    [
+        ("poly", PolynomialFeatures(degree=4)),
+        ("linear", LinearRegression(fit_intercept=False)),
+    ]
+)
+polyn_model_quant = Pipeline(
+    [
+        ("poly", PolynomialFeatures(degree=4)),
+        ("linear", QuantileRegressor(
+            alpha=1e-9,
+            fit_intercept=False,
+        )),
+    ]
+)
+
+STRATEGIES = {
+    "jackknife": {"method": "base", "cv": -1},
+    "jackknife_plus": {"method": "plus", "cv": -1},
+    "jackknife_minmax": {"method": "minmax", "cv": -1},
+    # "cv": {"method": "base", "cv": 10},
+    "cv_plus": {"method": "plus", "cv": 10},
+    "cv_minmax": {"method": "minmax", "cv": 10},
+    "quantile": {"method": "quantile", "cv": "split"},
```

Comment:
Rename the method to "conformalized_quantile_regression" ?

Reply:
done